text/0043-better-eviction.md

```diff
@@ -0,0 +1,126 @@
+happens with a delay:
+
+In Highway, validators take turns at proposing blocks, and the sequence of proposers is determined
+pseudorandomly. It is important that an attacker cannot manipulate the system in a way that allows
+them to choose among a large number of proposer sequences and thus get more slots. That is why
+the validator weights are determined in advance, with an `auction_delay` between the auction and
+the beginning of the era: A small change to any weight would change the proposer sequence for
+everyone, so it is important that before the auction, while bids can still change, the PRNG seed is
+not yet known.
+
+After the auction, during that delay, every block adds a single bit to the random seed that
+will be used to initialize the PRNG. That way, an attacker, even if they had the last three blocks
+before the new era's beginning, can choose only among eight different seeds, which are
+all likely to be fair. If they could still e.g. reduce two bids with a million tokens each, they
+could choose among a trillion different proposer sequences. Similarly, if an attacker has k
+validator nodes and could remove themselves by letting any subset of nodes exhibit a fault,
+they could choose among 2<sup>k</sup> leader sequences, even if those k nodes' stakes are small.
+
+We propose a safe way to effectively ignore faulty validators anyway, even if the fault happened
+after the auction, making it easier for the network to remain live without them.
+
+
+## Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+The PRNG seed for an era is still determined by the block in which the auction took place.
+And as a first step to compute the proposer sequence, that seed is still applied to the original
+validator weights.
+
+If after the auction a validator equivocates, it is still set to `Banned` status when the era is
+initialized. In addition, we now also do this if the validator was inactive. I.e. whenever we know
+that a validator is about to be evicted, we mark them as banned.
+
+Now the actual proposer sequence is modified:
+* If the proposer computation with the original set of weights returns a validator that was not
+  banned, the validator retains their slot.
+* If the computation returns a banned validator, the slot is reassigned by running the proposer
+  computation with the modified set of validators, with all banned ones removed.
+
+That way an attacker cannot gain more slots by making their nodes misbehave:
+They only lose some of their own slots.
+
+
+## Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+The first change to the protocol is that in addition to the equivocators, anyone who was reported
+as inactive in a switch block since the auction is marked as banned as well, when a new era is
+initialized. We will apply this only the the past `auction_delay` eras: If someone exhibited a fault
```

Comment:
```suggestion
initialized. We will apply this only to the past `auction_delay` eras: If someone exhibited a fault
```

Reply:
Done in https://github.com/casper-network/ceps/pull/43/commits/b31202ed858d95fd7338741c07cf45582dd4dbaa.